haskell-src/Concordium/Genesis/Data/BaseV1.hs

```diff
@@ -58,3 +65,25 @@ instance Serialize RegenesisDataV1 where
         genesisTerminalBlock <- get
         genesisStateHash <- get
         return RegenesisDataV1{..}
+
+-- |Convert 'GenesisParametersV3' to genesis data.
+-- This is an auxiliary function since much of the behaviour is shared between protocol versions.
+parametersToState :: GenesisParametersV3 pv -> (CoreGenesisParametersV1, GenesisState pv)
+parametersToState GenesisParametersV3{..} =
+    (CoreGenesisParametersV1{..}, GenesisState{..})
+  where
+    genesisTime = gp3GenesisTime
+    genesisEpochDuration = gp3EpochDuration
+    genesisLeadershipElectionNonce = gp3LeadershipElectionNonce
+    genesisAccounts = Vec.fromList gp3InitialAccounts
+    genesisCryptographicParameters = gp3CryptographicParameters
+    genesisIdentityProviders =
+        case filter (\(k, v) -> k /= ipIdentity v) (Map.toList (idProviders gp3IdentityProviders)) of
+            [] -> gp3IdentityProviders
+            ips -> error $ "Inconsistent identity provider ids: " ++ show ips
+    genesisAnonymityRevokers =
+        case filter (\(k, v) -> k /= arIdentity v) (Map.toList (arRevokers gp3AnonymityRevokers)) of
+            [] -> gp3AnonymityRevokers
+            ars -> error $ "Inconsistent anonymity revoker ids: " ++ show ars
```

Comment:
I think I had this comment in a previous PR already, but checking these seems arbitrary here.

For example we don't check that the anonymity revokers and identity providers that appear in the accounts are the right ones either.

So I think checking this is arbitrary and unclear why we need to bother.